### PR TITLE
[Slim] Refactoring

### PR DESCRIPTION
**Pull request by ybelenko**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.1.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

I found beautiful `AbstractPhpCodegen` class. In this pull request `PhpSlimServerCodegen` gets new config options and extends `AbstractPhpCodegen`. Now Slim codegen much smaller and cleanier.
Also I noticed old LICENSE file in Slim samples. Current codegen doesn't create LICENSE file for Slim, so I've deleted it.

cc @jebentier @dkarlovi @mandrean @jfastnacht @ackintosh

### CODE REVIEW

**Comment by ybelenko** [comment]
@ackintosh Don't merge this please, I've noticed #138 and want to add urgent changes to solve it.

**Comment by ybelenko** [comment]
@ackintosh Now I'm satisfied with the result. Please, review and merge when possible.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSlimServerCodegen.java [hunk 1]
```diff
@@ -60,62 +47,37 @@ public PhpSlimServerCodegen() {
...
-        typeMapping.put("map", "map");
-        typeMapping.put("array", "array");
-        typeMapping.put("list", "array");
-        typeMapping.put("object", "object");
-        typeMapping.put("binary", "\\SplFileObject");
+        // additionalProperties.put(CodegenConstants.ARTIFACT_VERSION, artifactVersion);
+        
+        // override cliOptions from AbstractPhpCodegen
+        for (CliOption co : cliOptions) {
+            if (co.getOpt() == AbstractPhpCodegen.VARIABLE_NAMING_CONVENTION) {
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSlimServerCodegen.java]: @ybelenko shall we use `.equals` instead for string comparison?

**Comment by ybelenko** [comment]
@ackintosh @wing328 Both recommendations has been applied

**Comment by wing328** [comment]
The shippable failure has been fixed in the master.

**Pull request merged**